File: frontend/react-website/src/types/Measurement.ts
export type MeasurementSlim = {
  id: number;
  name: string;
  unit: string;
};

export type Measurement = MeasurementSlim & {
  measurementPoints: Array<MeasurementPoint>;
};

export type MeasurementPost = {
  name: string;
  unit: string;
};

export type MeasurementPut = {
  id: number;
  name: string;
  unit: string;
};

export type MeasurementPoint = {
  id: number;
  value: number;
  createdAt: string;
};

export type MeasurementPointPost = {
  value: number;
};

File: backend/src/main/kotlin/tom/com/workout/logbook/model/measurement/Measurement.kt
package tom.com.workout.logbook.model.measurement

import jakarta.persistence.*
import org.hibernate.annotations.OnDelete
import org.hibernate.annotations.OnDeleteAction
import org.hibernate.annotations.UuidGenerator
import tom.com.workout.logbook.model.MetricFormat
import tom.com.workout.logbook.model.user.UserOwned
import java.util.*

@Entity
class Measurement(
    @Id
    @GeneratedValue
    @UuidGenerator
    var id: UUID? = null,
    var name: String,
    override var userId: UUID? = null,
    @Enumerated(EnumType.STRING)
    var metric: MetricFormat,

    @OneToMany(mappedBy = "measurement", cascade = [CascadeType.ALL])
    @OnDelete(action = OnDeleteAction.CASCADE)
    @OrderBy("createdAt ASC")
    var measurementPoints: List<MeasurementPoint> = emptyList()
) : UserOwned
